ui/src/app/abstract.service.ts

```diff
@@ -21,12 +21,12 @@ interface ApiGetParams {
 
 export abstract class AbstractService {
 
-  constructor(protected httpClient: HttpClient,
-              protected authService: IAuthService,
-              protected router: Router,
-              protected location: Location
-  )
-  {
+  protected constructor(
```

Comment:
This constructor, it's only used by child classes, also since it's an abstract class I think we don't need a public constructor, since in theory we should not create an instance of an abstract class.

But we can revert the change.